src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_broodlord_lashlayer.cpp

```diff
@@ -113,7 +121,85 @@ class boss_broodlord : public CreatureScript
     }
 };
 
+class go_suppression_device : public GameObjectScript
+{
+    public:
+        go_suppression_device() : GameObjectScript("go_suppression_device") { }
+
+        struct go_suppression_deviceAI : public GameObjectAI
+        {
+            go_suppression_deviceAI(GameObject* go) : GameObjectAI(go), _instance(go->GetInstanceScript()) { }
+
+            void InitializeAI() override
+            {
+                _events.ScheduleEvent(EVENT_SUPPRESSION_CAST, Seconds(0), Seconds(5));
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (_instance->GetBossState(DATA_VAELASTRAZ_THE_CORRUPT) != DONE)
+                    return;
+
+                if (_instance->GetBossState(DATA_BROODLORD_LASHLAYER) == DONE)
+                {
+                    if (me->GetGoState() != GO_STATE_DESTROYED)
+                    {
+                        me->SetGoState(GO_STATE_DESTROYED);
+                        me->SetFlag(GAMEOBJECT_FLAGS, GO_FLAG_NOT_SELECTABLE);
+                    }
+                    return;
+                };
+
+                if (me->GetGoState() == GO_STATE_DESTROYED)
+                    me->SetLootState(GO_READY); // NOT doing this will despawn the gameobject after using Disarm Trap
+
+                _events.Update(diff);
+
+                while (uint32 eventId = _events.ExecuteEvent())
+                {
+                    switch (eventId)
+                    {
+                        case EVENT_SUPPRESSION_CAST:
+                            if (me->GetGoState() == GO_STATE_READY)
+                            {
+                                me->SendCustomAnim(0);
+                                me->CastSpell(nullptr,SPELL_SUPPRESSION_AURA,true);
+                            }
+                            _events.ScheduleEvent(EVENT_SUPPRESSION_CAST, Seconds(5));
```

Comment:
```suggestion
                            _events.ScheduleEvent(EVENT_SUPPRESSION_CAST, 5s);
```

Reply:
Doesnt changes like this create a std::chrono overload warning?

Reply:
No, they are exactly the same type. 

The literal is defined like something along these lines:
```c++
constexpr Seconds operator ""s(unsigned long long s)
{
    return Seconds(s);
}
```
What wouldn't work is using a type with greater precision than milliseconds, since the `ScheduleEvent` accepts `Milliseconds` as parameter. In that case `std::chrono::duration_cast` has to be used to force the conversion to the less precise type.

Reply:
Thanks for that explanation :)